var/spack/packages/flux/package.py

```diff
@@ -0,0 +1,36 @@
+from spack import *
+import os
+
+class Flux(Package):
+    """ A next-generation resource manager (pre-alpha) """
+
+    homepage = "https://github.com/flux-framework/flux-core"
+    url      = "https://github.com/flux-framework/flux-core"
+
+    version('master', branch='master', git='https://github.com/flux-framework/flux-core')
+
+    # Also needs autotools, but should use the system version if available
+    depends_on("zeromq@4.0.4:")
+    depends_on("czmq@2.2:")
+    depends_on("lua@5.1:5.1.99")
```

Comment:
Just "@5.1" should match any 5.1.x.  It should look at that and try to match it against known versions, then pick one.  So if you depend on lua@5.1, the intended semantics is to take the highest available 5.1.  In the future that should change to "preferred 5.1", and packages and config files should be able to say which one they prefer.